### ISSUE TITLE
Updating AndroidView layer tree cause 100% crash on some brand device

### ISSUE THREAD

**Issue by qypengzai** [flutter/flutter#26345]
Hello,I met a crash when I use `AndroidView.When` I first time create `AndroidView(full screen)`, Everything works well. But when I try to create another `AndroidView`, adding them into a stack  and changing their layout(The new one is under the old one，this time the native method 'create' will recall twice because of the widget life cycle), it result in 100% crash.

this problem only occur in some device. I met twice: `VIVO NEX A and NUBIA NX595J`.

Here is the crash stack and log.
```
I/CatcherGestureDetector(14536): DecorView onAttached
I/CatcherGestureDetector(14536): window type filtered
I/View    (14536): updateSystemUiVisibility, blackBackground = true, getVisibility() = 0, getWindowVisibility() = 0, viewBounds = Rect(0, 0 - 0, 0), windowBounds = Rect(0, 0 - 0, 0), view = TextureView:android.view.TextureView{c150802 V.ED..... ......ID 0,0-0,0}
I/View    (14536): updateSystemUiVisibility, blackBackground = true, getVisibility() = 0, getWindowVisibility() = 0, viewBounds = Rect(0, 0 - 0, 0), windowBounds = Rect(0, 0 - 0, 0), view = TextureView:android.view.TextureView{c150802 V.ED..... ......ID 0,0-0,0}
I/View    (14536): updateSystemUiVisibility, blackBackground = true, getVisibility() = 0, getWindowVisibility() = 0, viewBounds = Rect(0, 0 - 1080, 2009), windowBounds = Rect(0, 0 - 1080, 2009), view = TextureView:android.view.TextureView{c150802 V.ED..... ......ID 0,0-1080,2009}
I/DecorView(14536): onConfigurationChanged mBackgroundPadding = Rect(0, 0 - 0, 0), mFramePadding = Rect(0, 0 - 0, 0), pkg = com.zego.flutter.videotalk
I/CatcherGestureDetector(14536): DecorView onAttached
I/CatcherGestureDetector(14536): window type filtered
I/View    (14536): updateSystemUiVisibility, blackBackground = true, getVisibility() = 0, getWindowVisibility() = 0, viewBounds = Rect(0, 0 - 0, 0), windowBounds = Rect(0, 0 - 0, 0), view = TextureView:android.view.TextureView{f13513 V.ED..... ......ID 0,0-0,0}
I/View    (14536): updateSystemUiVisibility, blackBackground = true, getVisibility() = 0, getWindowVisibility() = 0, viewBounds = Rect(0, 0 - 0, 0), windowBounds = Rect(0, 0 - 0, 0), view = TextureView:android.view.TextureView{f13513 V.ED..... ......ID 0,0-0,0}
I/View    (14536): updateSystemUiVisibility, blackBackground = true, getVisibility() = 0, getWindowVisibility() = 0, viewBounds = Rect(0, 0 - 351, 498), windowBounds = Rect(0, 0 - 351, 498), view = TextureView:android.view.TextureView{f13513 V.ED..... ......ID 0,0-351,498}


D/AndroidRuntime(14536): Shutting down VM
E/AndroidRuntime(14536): FATAL EXCEPTION: main
E/AndroidRuntime(14536): Process: com.zego.flutter.videotalk, PID: 14536
E/AndroidRuntime(14536): java.lang.NullPointerException: Attempt to invoke virtual method 'int android.view.Display.getDisplayId()' on a null object reference
E/AndroidRuntime(14536): 	at android.view.ViewRootImpl$1.onDisplayChanged(ViewRootImpl.java:1124)
E/AndroidRuntime(14536): 	at android.hardware.display.DisplayManagerGlobal$DisplayListenerDelegate.handleMessage(DisplayManagerGlobal.java:555)
E/AndroidRuntime(14536): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime(14536): 	at android.os.Looper.loop(Looper.java:192)
E/AndroidRuntime(14536): 	at android.app.ActivityThread.main(ActivityThread.java:6809)
E/AndroidRuntime(14536): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(14536): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
E/AndroidRuntime(14536): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:817)
```

Here is the doctor output
```
lizhanpengZEGOde-MacBook-Pro:flutter_videotalk lizhanpeng$ flutter doctor -v
[✓] Flutter (Channel dev, v1.1.7, on Mac OS X 10.14 18A391, locale zh-Hans-CN)
    • Flutter version 1.1.7 at /Users/lizhanpeng/flutter
    • Framework revision 0d5ee0560b (2 days ago), 2019-01-08 09:24:27 -0800
    • Engine revision 7112b72cc2
    • Dart version 2.1.1 (build 2.1.1-dev.0.1 ec86471ccc)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/lizhanpeng/Library/Android/sdk
    • Android NDK at /Users/lizhanpeng/Library/Android/sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/lizhanpeng/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[!] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    ✗ Verify that all connected devices have been paired with this computer in Xcode.
      If all devices have been paired, libimobiledevice and ideviceinstaller may require updating.
      To update with Brew, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew uninstall --ignore-dependencies usbmuxd
        brew install --HEAD usbmuxd
        brew unlink usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 2.0.0
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 28.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[!] IntelliJ IDEA Ultimate Edition (version 2018.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[✓] VS Code (version 1.30.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.21.1

[✓] Connected device (1 available)
    • vivo NEX A • 9cb87f39 • android-arm64 • Android 8.1.0 (API 27)
```

**Comment by qypengzai** [comment]
When I use vivo NEX A to run google map exmple demo, trying updating its layer tree, it also result in crash.

**Comment by zoechi** [comment]
>  also result in crash.

with exactly the same error output?

**Comment by qypengzai** [comment]
@zoechi all the crash stack are same, today Google Pixel also result in this crash when I update the view layer rapidly(like added view and removed view rapidly). it is not 100% crash.

**Comment by qypengzai** [comment]
@zoechi Thanks for rapid reply.I really wish that this crash will be solved as soon as possible because we have released our product online.

**Comment by kangwang1988** [comment]
@qypengzai 
A minimal demo project which can reproduce this issue would be much helpful.

**Comment by kangwang1988** [comment]
@qypengzai 
I would suggest you to create a minimal demo project on github.

**Comment by qypengzai** [comment]
ok,when I finish , I'll get back to you. Thank you very much.

**Comment by qypengzai** [comment]
@kangwang1988 ok,when I finish , I'll get back to you. Thank you very much.

**Comment by gopchi** [comment]
Did anyone resolve this issue? We are encountering it on many devices and keeps our main functionality of the Google map broken...

Please if someone has any idea how to fix it even locally would be also a thing

**Comment by swat-cat** [comment]
Hello, same issue using OnePlus, Xiaomi with google maps - java.lang.NullPointerException: Attempt to invoke virtual method 'int android.view.Display.getDisplayId()' on a null object reference

**Comment by ikakaxi** [comment]
I also encountered this problem. My phone model is 360 N6 Pro.

**Comment by davidgalarza** [comment]
Any update? Please take action on this

**Comment by TheDOme6** [comment]
confirmed on Huawei P20 Lite Android 8.0.0

**Comment by yuridiniz** [comment]
Also happening on the Firebase Test Lab's Pixel 2 API 26 (Android 8.0) device, it's also happening on 11 devices, all using Android 8.1 (API 27)

According to the google play report, it happened in these devices

```
Moto G (5th Gen) (cedric)
Moto G (5S) Plus (sanders_nt)
ZenFone Max Pro (M1) (ASUS_X00T_2)
C5 (C014)
moto e5 play (pettyl)
Moto G Plus (5th Gen) (potter_nt)
Moto G (5S) (montana)
B2017G (tulip)
ZenFone Max Shot (ZB634KL) (ASUS_A001D_2)
Moto G(4) Plus (athene_f)
ZenFone Max Plus M2 (ZB634KL) (ASUS_A001D_1)

```

**Comment by rpk98c** [comment]
Same issue on Pixel 3 Android 9... any update on a fix. Blocker for us as maps is a major part of the app.

**Comment by TheDOme6** [comment]
This mostly happens, when I rapidly call setState.

**Comment by impe93** [comment]
I made a custom AndroidView and the first time that I navigate to the page where it is, it doesn't crash, but when I try to navigate for the second time to the same page the custom view make the app crash with the error reported in this issue. I tested it on P20 lite with Android 8. After I tried on an Honor 8 with Android 8 and everything works fine everytime.

**Comment by melbit-lukeshan** [comment]
Hope this help. In our app we found similar issue. Maybe 60% time the app will crash when we get in/out the map page. It looks like the issue is related to the Hero animation. After removing the Hero animation we didn't encounter the issue any more.

**Comment by nemoryoliver** [comment]
> This mostly happens, when I rapidly call setState. Occurs on the admob_flutter plugin.

This is exactly the same thing for me.

**Comment by gopchi** [comment]
@melbit-lukeshan @nemoryoliver @TheDOme6 
In our case we do not setState and do not have Hero animations and it crashes constantly even with the latest flutter update v1.5.4-hotfix.2. Will anyone from the Google team ever look at this? How can you have such a great Google product like Flutter that cannot use another Google product like Google Maps, with which people want to build apps upon. 
@zoechi

**Comment by kbrmimbyl** [comment]
Having this issue with Moto G (5S) Plus right now. The crashes appear random.

**Comment by kbrmimbyl** [comment]
> This mostly happens, when I rapidly call setState. Occurs on the admob_flutter plugin.

Rapidly?

**Comment by TheDOme6** [comment]
Tested it by calling setState every 0.5 seconds

**Comment by Eddydpyl** [comment]
Found the same issue in my app, in my case there are two likely suspects:
- A call to setState throwing an error, due to me calling the method when I shouldn't have.
- The [Background Geolocation library](https://github.com/transistorsoft/flutter_background_geolocation).

In the first case, I think it might me the cause because the error was thrown at the same time as the app crashed, and solving it appears to have fixed this issue. As for the second case, it was the library itself the one that threw the error from within my app, rather than my own code, or so it seemed. It's quite tricky for me to pinpoint the error, as it's very difficult to reproduce.

**Comment by apgapg** [comment]
Even i have to hide google maps in my app due to this issue. I hope team fixes it soon. There's really no way out for now.

**Comment by trantrongkim98** [comment]
who can fix this issue?

**Comment by qypengzai** [comment]
@zoechi I hope flutter develop team will fix this issue as soon as possible.

**Comment by chunhtai** [comment]
@qypengzai do you have a reproducible example?

**Comment by kagile** [comment]
> > also result in crash.
> 
> with exactly the same error output?

@zoechi Any update on this?...

@qypengzai  Trying google maps plugin crashes with the same exception.

**Comment by rafaelcorbellini-egsys** [comment]
Same error here, when i re-enter fast on some page with maps in Xiomi or Moto G

![WhatsApp Image 2019-08-13 at 15 25 34](https://user-images.githubusercontent.com/49765550/62970910-e72b1980-bde6-11e9-8a6a-2d4e450f00a2.jpeg)

**Comment by trantrongkim98** [comment]
who has information on this issue?

**Comment by trantrongkim98** [comment]
My google_map still crashes, hopefully there is an explanation for this issue

**Comment by TryLoveCatch** [comment]
me too

**Comment by iiinaiii** [comment]
I encountered the same crash using google_maps_flutter.
An error occurred when go to the second google_map page from a screen with a keyboard.

I fixed it to navigate to the google_map page after hide the keyboard.

(Phone model is Xperia XZ3 SO-01L)

**Comment by chenzongwen** [comment]
> Hi, this crash occurs in our apps also, but only on newer, taller phones like xiaomi 9.
> I found that if the height/width ratio is bigger crash happens
> 
> crash:
> 
> ```
> return Container(
>   width: MediaQuery.of(context).size.width,
>   height: MediaQuery.of(context).size.width * 1.8,
>   child: GoogleMap(
> 	  initialCameraPosition: CameraPosition(
> 		  target: const LatLng(50.0, 19.0), zoom: 3.0)),
> );
> ```
> 
> doesn't crash:
> 
> ```
> return Container(
>   width: MediaQuery.of(context).size.width,
>   height: MediaQuery.of(context).size.width * 1.5,
>   child: GoogleMap(
> 	  initialCameraPosition: CameraPosition(
> 		  target: const LatLng(50.0, 19.0), zoom: 3.0)),
> );
> ```

have you find the solution?

**Comment by mayurdhurpate** [comment]
I'm also getting a similar error from multiple users from Xiaomi
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'int android.view.Display.getDisplayId()' on a null object reference
       at android.view.ViewRootImpl$1.onDisplayChanged(ViewRootImpl.java:1145)
       at android.hardware.display.DisplayManagerGlobal$DisplayListenerDelegate.handleMessage(DisplayManagerGlobal.java:493)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:171)
       at android.app.ActivityThread.main(ActivityThread.java:6656)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:873)
```

Another similar kind of error which I'm getting:

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.DisplayAdjustments android.view.Display.getDisplayAdjustments()' on a null object reference
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1761)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1505)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7354)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1092)
       at android.view.Choreographer.doCallbacks(Choreographer.java:888)
       at android.view.Choreographer.doFrame(Choreographer.java:819)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1078)
       at android.os.Handler.handleCallback(Handler.java:873)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:201)
       at android.app.ActivityThread.main(ActivityThread.java:6810)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:873)
```

on many production devices. We only use https://github.com/sarbagyastha/youtube_player_flutter for embedding a few YouTube videos and some logs of Crashylytics points to users opening it before the crash, but not many data points to confirm anything. Was unable to reproduce the bug on my local Android/iOS device and emulator.

**Comment by zorozhao** [comment]
me too

**Comment by PingShen** [comment]
me too

**Comment by sergicastellsague** [comment]
@mklim is there any chance this work can be started anytime soon? this one plus https://github.com/flutter/flutter/issues/38063 are the two ones causing most of our crashes

**Comment by iiinaiii** [comment]
This is a workaround solution for a crash when using google_maps_flutter on my app.

Before building the Flutter widget, I had to wait for GoogleMap to be ready, which would be displayed in PlatformView. After the onMapCreated callback, wait 100msec before building the Flutter Widget.

Certain widgets seemed to affect the crash. In my case,  I delayed BottomAppBar build.

---

Code sample:
```
class _TopPageState extends State<TopPage> {

  bool _mapReady = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: GoogleMap(
                onMapCreated: (controller) => _onMapCreated(),
              ),
      bottomNavigationBar: _mapReady ? _buildBottomAppBar(context) : null,
    );
  }

  _onMapCreated() {
    Future.delayed(const Duration(milliseconds: 100), () {
      setState(() {
        _mapReady = true;
      });
    });
  }
}
```

**Comment by mklim** [comment]
Unfortunately I don't think it likely that I'm going to be able to work on this in the near term future, so I'm taking off my assignment and resetting the milestone. Sorry about that everyone. :disappointed: 

For what it's worth for anyone else looking into this I had started investigating this briefly already, and I was blocked on not being able to get reproduction of the crash. I had tried with some of the devices mentioned in this report, but not all of them. I haven't tried the most recent code sample posted here either.

**Comment by singun-lxd** [comment]
@mklim I have demo to reproduce this crash, 100%.

When WebView is **playing videos**, when you open the **keyboard**, the crash happen immediately.

You can find my demo in https://github.com/singun-lxd/flutter_webview_crash, or another simple demo below.

Models:
Huawei Y9 2019, Android 9
Huawei nova 3i, Android 9

Stack:
```
2020-03-04 13:36:08.938 3901-3901/com.demo.flutter_app E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.demo.flutter_app, PID: 3901
    java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.DisplayAdjustments android.view.Display.getDisplayAdjustments()' on a null object reference
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1900)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1636)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7946)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1092)
        at android.view.Choreographer.doCallbacks(Choreographer.java:893)
        at android.view.Choreographer.doFrame(Choreographer.java:812)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1078)
        at android.os.Handler.handleCallback(Handler.java:907)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:216)
        at android.app.ActivityThread.main(ActivityThread.java:7625)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
```

Demo code:

```
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter WebView Crash'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(title),
      ),
      body: Column(
        children: <Widget>[
          Text("Please tap the text field below to trigger the crash"),
          TextField(),
          Expanded(
            child: WebView(
              initialUrl: "https://www.youtube.com/watch?v=mlBIpwtNxxE",
              javascriptMode: JavascriptMode.unrestricted,
            ),
          ),
        ],
      ),
    );
  }
}
```

Dependeccies:
```
webview_flutter: 0.3.19+8
```

**Comment by GWesley** [comment]
Got a crash report for nubia 563j with OS 7.1.1. Our app has a google map view on the homepage.

**Comment by edTheGuy00** [comment]
I found this issues with the `google_maps_flutter` plugin. My workaround was just to wrap it in a `FutureBuilder` with a 500ms delay.

```dart
              return FutureBuilder(
                  future:
                      Future.delayed(Duration(milliseconds: 500), () => true),
                  builder: (ctx, future) {
                    if (future.hasData) {
                      return Container(
                        child: GoogleMap(
                          mapToolbarEnabled: false,
                          myLocationButtonEnabled: false,
                          markers: snapshot.data,
                          mapType: MapType.normal,
                          initialCameraPosition: camPos,
                          onCameraMove: _homeBloc.streams.listenToMapMovement,
                          onMapCreated: _homeBloc.registerMapController,
                        ),
                      );
                    } else {
                      return Center(
                        child: SpinKitCubeGrid(
                          color: Colors.white70,
                        ),
                      );
                    }
                  })
```

**Comment by chhota89** [comment]
Any solution for this. I am using the online video call library in that also same error is coming

**Comment by GaryQian** [comment]
Repro-ed on Huawei Nova 4 (VCE-AL00), thanks @singun-lxd for the straightforward repro!

**Comment by dotcink** [comment]
I've submit a PR(flutter/engine/pull/17611) to Flutter Engine repo.

To fix it currently without having to wait for PR acceptance and flutter release, I use **AOP** to add some code to the flutter engine class(very simply, with only a couple of lines of code).

**Here is the PATCH file**: https://raw.githubusercontent.com/dotcink/flutter_webview_crash/fix-crash/fix.patch

It's been tested with @singun-lxd 's repro project, as in https://github.com/dotcink/flutter_webview_crash/tree/fix-crash.

**Comment by tonyclickspace** [comment]
> To fix it currently without having to wait for PR acceptance and flutter release, I use **AOP** to add some code to the flutter engine class(very simply, with only a couple of lines of code).

@dotcink Can you provide instructions on using AOP to test this out? I'm not familiar with it, and I'd like to test if this clears out #53598 as well.

**Comment by dotcink** [comment]
> > To fix it currently without having to wait for PR acceptance and flutter release, I use **AOP** to add some code to the flutter engine class(very simply, with only a couple of lines of code).
> 
> @dotcink Can you provide instructions on using AOP to test this out? I'm not familiar with it, and I'd like to test if this clears out #53598 as well.

You can see the diff at https://github.com/dotcink/flutter_webview_crash/commit/0bd6d2ac46022accfaee6b6663c2c5f6aa4ae9d8

Or just try to apply the patch https://raw.githubusercontent.com/dotcink/flutter_webview_crash/fix-crash/fix.patch 

It's very simple. No more than 20 lines of code.

**Comment by GaryQian** [comment]
This should be fixed by https://github.com/flutter/engine/pull/17897

**Comment by mmahkamov** [comment]
Is there any chance the fix could land in a hotfix for 1.12.13?

This is a number one crash in our app's reports and number one complaint by our users with Xiaomi phones.

**Comment by GaryQian** [comment]
The changes have landed in the engine. Once it rolls into the framework (should happen in next few hours), it would be helpful if someone could verify it fixes the crash in their specific case.

Once verified, I can see if this is hotfixable. I will submit it to the hotfix list.

**Comment by tonyclickspace** [comment]
> [...]it would be helpful if someone could verify it fixes the crash in their specific case.

Confirmed, it fixes [the crash in singun-lxd's Feb 24 comment](#issuecomment-590685810), as well as the one documented at #53598. Tested on both the DragonTouch M7 tablet, and the Minix NEO T5 Android TV.

**Comment by qimingweng** [comment]
How can we tell if this fix, (or specifically flutter/engine#17897) is merged into dev or beta or a hotfix right now? The changelogs are not super helpful.

**Comment by mklim** [comment]
@qimingweng it's a little involved. There's a guide on the ["Where's my commit?"](https://github.com/flutter/flutter/wiki/Where%27s-my-Commit%3F#finding-the-framework-commit-that-contains-engine-commit-x) page on the Wiki for tracking specific commits like this.

In this case it's rolled into the framework in flutter/flutter@e0c63cd35e15e407a80dc44281cc392535fcce25, `v1.18.0-8.0.pre`. which you can then see on [the SDK releases page](https://flutter.dev/docs/development/tools/sdk/releases) has currently rolled into the `dev` but no other channel.

**Comment by qimingweng** [comment]
Oh wow OK. @mklim I think I follow but the instructions are a bit involved haha. Thanks for pointing it out, really appreciated it.

For everyone else, I can verify that this (the dev channel) did fix the crash on google_maps_flutter on a Nexus 5X device on Android 8.1.

**Comment by qimingweng** [comment]
Specifically @GaryQian can confirm this fixes one symptom of the issue and would appreciate it if we could hotfix this.

**Comment by GaryQian** [comment]
@qimingweng I will see what I can do!

**Comment by qimingweng** [comment]
Hey, just for record's sake, we upgraded to Flutter 1.17.0 and the issue seems to have gone away on our Nexus 5X for Google Maps. However, I also went through the process from this link: https://github.com/flutter/flutter/wiki/Where%27s-my-Commit%3F#finding-the-framework-commit-that-contains-engine-commit-x and could not necessarily confirm that this fix has rolled into 1.17.0 (no hotfix suffix).

For whatever it's worth.

**Comment by chhota89** [comment]
I have tested this issue with the dev channel. This issue is resolved in the dev channel.

**Comment by qimingweng** [comment]
We noticed there is a new version of flutter, 1.17.1 out May 13. Do those tags indicate that this issue is resolved in this version?

**Comment by renefloor** [comment]
@qimingweng yes, it was mentioned in the release notes: https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel#flutter-117-changes

**Comment by qimingweng** [comment]
For what it's worth, I can confirm this has fixed our issue in stable-1.17.1.
